src/views/catalog/wizard/tabs/scripts/components/CloudInitNetworkForm.tsx

```diff
@@ -0,0 +1,199 @@
+      }
+    });
+  };
+
+  const onUpdateVM = React.useCallback(
+    () =>
+      updateVM((vmToUpdate) =>
+        produceVMDisks(vmToUpdate, (vmDraft) => {
+          const cloudInitDiskName = cloudInitVolume?.name || 'cloudinitdisk';
+          const cloudInitDisk = vmDraft.spec.template.spec.domain.devices.disks.find(
+            (disk) => disk.name === cloudInitDiskName,
+          );
+
+          // cloudinitdisk deleted or doesn't exist, we need to re-create it
+          if (!cloudInitDisk) {
+            vmDraft.spec.template.spec.domain.devices.disks.push({
+              name: cloudInitDiskName,
+              disk: {
+                bus: 'virtio',
+              },
+            });
+          }
+
+          const updatedCloudinitVolume = {
+            ...(cloudInitVolume || { name: cloudInitDiskName }),
+            cloudInitNoCloud: {
+              ...(cloudInitVolume?.cloudInitNoCloud || {}),
+              ...CloudInitDataHelper.toCloudInitNoCloudNetworkSource(dump(networkData), isBase64),
+            },
+          };
+
+          const otherVolumes = getVolumes(vmDraft).filter((vol) => !vol.cloudInitNoCloud);
+          vmDraft.spec.template.spec.volumes = [...otherVolumes, updatedCloudinitVolume];
+        }),
+      ),
+    [cloudInitVolume, isBase64, updateVM, networkData],
+  );
+
+  React.useEffect(() => {
+    if (networkData) {
+      const ipAddreses = networkData.network.config[0].subnets[0].address.join(', ');
+      const hasChanged =
+        ipAddreses !== '' ||
+        networkData.network.config[0].subnets[0].gateway !== '' ||
+        networkData.network.config[0].name !== '';
+      const isValid =
+        validateIpAddresses(ipAddreses) &&
+        validateGateway(networkData?.network?.config[0]?.subnets?.[0]?.gateway);
+
+      const timerId = setTimeout(() => isValid && hasChanged && onUpdateVM(), 100);
+
+      return () => clearTimeout(timerId);
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [networkData]);
+
+  return (
+    <>
+      <FormGroup fieldId="divider">
+        <Divider />
+      </FormGroup>
+      <FormGroup fieldId="network-label" label={t('Network data')}></FormGroup>
+      <FormGroup
+        label={t('Ethernet name')}
+        fieldId={'ethernet-name'}
+        className="kv-cloudint-advanced-tab--validation-text"
+        helperText={t('Please provide name.')}
+      >
+        <TextInput
+          value={networkData?.network?.config[0]?.name || ''}
+          type="text"
+          id={'ethernet-name'}
+          onChange={(v) => onFieldChange(CloudInitNetworkFormKeys.NAME, v)}
+        />
+      </FormGroup>
+      <FormGroup
+        label={t('Addresses')}
+        fieldId={'address'}
+        className="kv-cloudint-advanced-tab--validation-text"
+        helperText={t('Please provide IP address.')}
+        helperTextInvalid={t('Please provide valid IP addresses, separated by comma.')}
```

Comment:
we don't know what valid address is, we also can't enforce comma as seperator for same reasons